loader/src/main.py

```diff
@@ -172,11 +174,26 @@ def process_event(msg):
             # references OR is referenced by other documents)
             logger.debug(
                 f"Resolving references {analysis.reference_paths}",
-                extra={"resource_id": resource_id},
+                extra={"batch_id": batch_id, "resource_id": resource_id},
+            )
+
+            # Redis distributed lock for reference binding
+            dlm = Redlock(
+                [{
+                    "host": REDIS_REFERENCES_HOST,
+                    "port": REDIS_REFERENCES_PORT,
+                    "db": REDIS_REFERENCES_DB
+                }, ]
             )
+            while True:
+                # Lock TTL is 5 min
+                ref_lock = dlm.lock(f"batch:{batch_id}:lock", 3600_000)
```

Comment:
`3600_000` -> `3_600_000`
I don't think it's equivalent to 5 minutes

Reply:
oops fixed